.github/workflows/e2e-deploy.yml

```diff
@@ -93,58 +125,307 @@ jobs:
+            -portforward-pod-name=${podname} \
+            -portforward-pod-port=8081 \
+            -kubeconfig=${KUBECONFIG}
+        env:
+          DATASET: fashion-mnist-784-euclidean.hdf5
+  e2e-deploy-sidecar:
+    name: "E2E deploy test (Agent & Sidecar)"
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+    if: startsWith( github.ref, 'refs/tags/') || github.event.action == 'labeled' && github.event.label.name == 'actions/e2e-deploy'
+    steps:
+      - uses: actions/checkout@v2
+      - name: wait for dockers
+        if: startsWith( github.ref, 'refs/tags/')
+        run: |
+          tag=$(cat versions/VALD_VERSION)
+          for image in \
+            vdaas/vald-agent-ngt \
+            vdaas/vald-agent-sidecar
+          do
+            echo "searching ${image}:${tag}"
+            until curl -s "https://registry.hub.docker.com/v2/repositories/${image}/tags/${tag}" | jq '.name' | grep -v "null"; do
+              echo "waiting for ${image}:${tag} to be uploaded..."
+              sleep 2
+            done
+          done
+      - name: Specify container versions
+        if: github.event.action == 'labeled' && github.event.label.name == 'actions/e2e-deploy'
+        run: |
+          pr_num=`cat $GITHUB_EVENT_PATH | jq -r ".number"`
+          if curl -s "https://registry.hub.docker.com/v2/repositories/vdaas/vald-agent-ngt/tags/pr-${pr_num}" | jq '.name' | grep -v "null"; then
+            export HELM_EXTRA_OPTIONS="${HELM_EXTRA_OPTIONS} --set agent.image.tag=pr-${pr_num}"
+          fi
+          if curl -s "https://registry.hub.docker.com/v2/repositories/vdaas/vald-agent-sidecar/tags/pr-${pr_num}" | jq '.name' | grep -v "null"; then
+            export HELM_EXTRA_OPTIONS="${HELM_EXTRA_OPTIONS} --set agent.sidecar.image.tag=pr-${pr_num}"
+          fi
+          echo "HELM_EXTRA_OPTIONS=${HELM_EXTRA_OPTIONS}"
+          echo "HELM_EXTRA_OPTIONS=${HELM_EXTRA_OPTIONS}" >> $GITHUB_ENV
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libhdf5-dev
+      - name: Fetch Helm version
+        run: |
+          HELM_VERSION=`make version/helm`
+          echo "::set-output name=helm::${HELM_VERSION}"
+        id: version
+      - uses: rinx/setup-k3d@v0.0.2
+        with:
+          version: latest
+          name: vald
+          agents: 3
+      - name: check k3d
+        run: |
+          kubectl cluster-info
+      - uses: azure/setup-helm@v1
+        with:
+          version: ${{ steps.version.outputs.helm }}
+      - name: Helm version
+        run: |
+          helm version
+      - name: deploy vald
+        run: |
+          kubectl apply -f k8s/external/minio
+          jq_query='.items[] | select( ([ .status.conditions[] | select( .type == "Ready" and .status == "True" ) ] | length ) == 1) | .metadata.namespace + "/" + .metadata.name'
+          until [ $(kubectl get pod --selector=app=minio -o json | jq -r "$jq_query" | wc -l) -ge 1 ]
```

Comment:
I think it would be better to wait in make command that i commented

Reply:
see this example 
https://github.com/vdaas/vald/blob/master/Makefile.d/k8s.mk#L266

Reply:
thanks!